src/agent/create-test-runner.js

```diff
@@ -36,11 +36,15 @@ export async function createRunner(options) {
       browser: options.webDriverBrowser,
       abortSignal: options.abortSignal,
       log: options.log,
+    }).catch(err => {
+      throw new Error(`Error connecting to web-driver: ${err}`);
```

Comment:
This will discard the original stack trace. I think `cause` is what we want here:

```suggestion
      throw new Error('Error connecting to web-driver', {cause: err});
```

Alternatively, modifying `err.message` and re-throwing it would also disambiguate while preserving the stack trace.

Reply:
awesome! I don't think I've ever seen this feature, I was looking around for the sort of equivalent to C# `new Exception(string message, Exception inner)` and couldn't find one or remember one.  I learned something new about JS today :) (how long has this been lurking there?)

> Alternatively, modifying `err.message` and re-throwing it would also disambiguate while preserving the stack trace.

I think the `{cause}` method is more useful, and in the event of some non-`Error` thrown error.